src/components/component-definition.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import ts from 'typescript';
+
+import { extractDeclaration, stringifyType } from './type-utils';
+import type {
+  ComponentDefinition,
+  ComponentFunction,
+  ComponentProperty,
+  ComponentRegion,
+  EventHandler,
+} from './interfaces';
+import type { ExpandedProp } from './extractor';
+import { getObjectDefinition } from './object-definition';
+
+function getCommentTag(property: ExpandedProp, name: string) {
+  const tag = property.description.tags.find(tag => tag.name === name);
+  return tag ? tag.text ?? '' : undefined;
+}
+
+function castI18nTag(tag: string | undefined) {
+  return tag === undefined ? undefined : true;
+}
+
+export function buildComponentDefinition(
+  name: string,
+  props: Array<ExpandedProp>,
+  functions: Array<ExpandedProp>,
+  defaultValues: Record<string, string>,
+  checker: ts.TypeChecker
+): ComponentDefinition {
+  const regions = props.filter(prop => prop.type === 'React.ReactNode');
+  const events = props.filter(prop => prop.name.match(/^on[A-Z]/));
+  const onlyProps = props.filter(prop => !events.includes(prop) && !regions.includes(prop));
+
+  return {
+    name,
+    releaseStatus: 'stable',
+    regions: regions.map(
+      (region): ComponentRegion => ({
+        name: region.name,
+        displayName: getCommentTag(region, 'displayname'),
+        description: region.description.text,
+        isDefault: region.name === 'children',
+        visualRefreshTag: getCommentTag(region, 'visualrefresh'),
+        deprecatedTag: getCommentTag(region, 'deprecated'),
+        i18nTag: castI18nTag(getCommentTag(region, 'i18n')),
```

Comment:
do we have any cases where there is i18n support for regions?

Reply:
Not sure about anywhere in the components, but in this repo we have a unit test for it

https://github.com/cloudscape-design/documenter/blob/5bc27096f251cdb38ea508b036dd8b21af49403a/fixtures/components/i18n-tag/example/index.tsx#L17-L22

So I decided I keep it